canton/community/ledger/ledger-common/src/main/scala/com/digitalasset/canton/ledger/error/PackageServiceErrors.scala

```diff
@@ -314,13 +314,35 @@ object PackageServiceErrors extends PackageServiceErrorGroup {
       )(implicit
           val loggingContext: ContextualizedErrorLogger
       ) extends DamlError(
-            cause = "A DAR with the same version number has previously been uploaded.",
-            extraContext = Map(
-              "uploadedPackage" -> uploadedPackage,
-              "existingPackage" -> existingPackage,
-              "packageVersion" -> packageVersion.toString,
-            ),
-          )
+          cause = "A DAR with the same version number has previously been uploaded.",
+          extraContext = Map(
+            "uploadedPackage" -> uploadedPackage,
+            "existingPackage" -> existingPackage,
+            "packageVersion" -> packageVersion.toString,
+          ),
+        )
+    }
+
+    @Explanation(
+      """This error indicates that the dry run flag was set in UploadDarRequest, so the uploaded Dar must not be uploaded."""
+    )
+    @Resolution("Upload with `dry_run: false` in the UploadDarRequest")
+    object DryRun
+        extends ErrorCode(
+          id = "DAR_DONT_UPLOAD_DUE_TO_DRY_RUN",
+          ErrorCategory.InvalidIndependentOfSystemState,
+        ) {
+      final case class Error(
+          uploadedPackage: Ref.PackageId,
+      )(implicit
+          val loggingContext: ContextualizedErrorLogger
+      ) extends DamlError(
+          cause =
+            "The DAR will not be uploaded because dry run is set in UploadDarRequest",
+          extraContext = Map(
+            "uploadedPackage" -> uploadedPackage,
```

Comment:
I'd probably prefer to call this `uploadedPackageId`

Reply:
fixed